design/tls-client-verification.md

```diff
@@ -0,0 +1,161 @@
+# Client certificate validation (mTLS)
+
+Status: Draft
+
+## Goals
+
+- Allow client certificate validation (mTLS) on contour (performed by Envoy)
+- Allow various ways for client certificate validation; Spki, Hash and CA
+- Document mTLS configuration
+
+## Non Goals
+
+- Support for k8s ingress documents.
+- Repeat configuration details described in Envoy documentation (make references)
+
+## Background
+
+In TLS (and https) only the server is authenticated with a
+certificate, for instance you as a client can be sure that you speak
+with your bank and not some malicious site. But sometimes also the client
+must be authenticated. As noted in
+[wikipedia](https://en.wikipedia.org/wiki/Mutual_authentication)
+client certification (mTLS) is not very common for end-users but is
+more widespread for business-to-business (B2B) applications (which
+may use gRPC and REST APIs).
+
+I can't give a real-life example but it is easy to imagine cases where
```

Comment:
I'd like to challenge this statement: "I can't give a real-life example but it is easy to imagine cases". 

You're looking to add this feature, so there's a need to have this, right? We should be adding real-life use-cases to back up the proposal, without those, then I question why we are adding this feature.

Reply:
I have updated the examples with the ones in the comment above https://github.com/heptio/contour/pull/1233#discussion_r306218224